### DISCUSSION TITLE
3.3.0 RC4

### DISCUSSION THREAD

**Discussion by sfmskywalker** [elsa-workflows/elsa-core#6214] [Design & tech]
## What's Changed
* Fix: Resolve Python.Runtime.BadPythonDllException in Elsa.ServerAndStudio.Web by @amirhaghajani in https://github.com/elsa-workflows/elsa-core/pull/6151
* fix JavaScript BigInt mapping to BigInteger serialization incorrect by @fanslead in https://github.com/elsa-workflows/elsa-core/pull/6164
* Enhance Multitenancy with Runtime Tenant Management and Task Handling   by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6173
* Fix JS Variable Setting Bug in Workflow Variable Sync Logic   by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6175
* Fix route data handling by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6176
* Refactor and reorganize Workflow Storage modules. by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6177
* Add synthetic properties to JS engine for exclusion by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6178
* Introduce job unscheduling functionality by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6183
* Fix race condition and add rate limiting to bookmark queue processing by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6187
* Add schema registry support to Kafka module by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6190
* Add TenantId to RunTaskWebhookPayload by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6191
* Persist workflow variables during background execution. by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6195
* Add .NET 9.0 target by @Suchiman in https://github.com/elsa-workflows/elsa-core/pull/6186
* Fix C# Script<T> race conditino by @Suchiman in https://github.com/elsa-workflows/elsa-core/pull/6192
* Fix Primitive Collection serialization by @Suchiman in https://github.com/elsa-workflows/elsa-core/pull/6157
* [Feat: OTEL + Tenants] : Add TenantId Property to Span  by @jdevillard in https://github.com/elsa-workflows/elsa-core/pull/6083
* Fix Quartz scheduler implementation by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6200
* Introduce the possibility to exclude variable value retrieval for certain storage drivers by @MariusVuscanNx in https://github.com/elsa-workflows/elsa-core/pull/6201
* Disable activity state filtering temporarily. by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6202
* Fixed variable value retrieval by @MariusVuscanNx in https://github.com/elsa-workflows/elsa-core/pull/6204
* Add concurrency locks to DefaultTenantService operations by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6205
* Regenerate EF Core migrations for 3.2 compat by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6206
* Improve variable parsing with error handling and logging by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6207
* Fixed issues where the outputs of a background activity will not be mapped by @MariusVuscanNx in https://github.com/elsa-workflows/elsa-core/pull/6209
* Refactor variable handling in storage driver logic by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6210
* Retention: Add sweep interval to feature configuration by @Sverre-W in https://github.com/elsa-workflows/elsa-core/pull/6208
* Retention: Allow to define cleanup strategy for workflow instances by @Sverre-W in https://github.com/elsa-workflows/elsa-core/pull/6211
* Refactor activity cancellation and bookmark handling. by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/6213

## New Contributors
* @amirhaghajani made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/6151
* @fanslead made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/6164

**Full Changelog**: https://github.com/elsa-workflows/elsa-core/compare/3.3.0-rc3...3.3.0-rc4

<hr /><em>This discussion was created from the release <a href='https://github.com/elsa-workflows/elsa-core/releases/tag/3.3.0-rc4'>3.3.0 RC4</a>.</em>

**Comment by ascendedguard** [comment]
Was RC4 published on NuGet? I'm only seeing RC3 for core, and RC4 for Studio with no way to install the pre-release for Studio since the RC4 API dependency cannot be found.

**Comment by sfmskywalker** [comment]
Yes. The `3.3.0-rc4` release was published on December 21, 2024, after this comment. The matching `Elsa.Api.Client` and `Elsa.Studio` `3.3.0-rc4` packages are available on NuGet now, so they can be restored together using that same prerelease version.

Closing this release discussion as resolved.
